File: sling-react-core/src/main/java/com/cathcart93/sling/services/BeanSerializer.java
package com.cathcart93.sling.services;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import com.fasterxml.jackson.databind.ser.BeanSerializerFactory;
import com.fasterxml.jackson.databind.ser.SerializerFactory;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import org.apache.felix.scr.annotations.Activate;
import org.apache.felix.scr.annotations.Component;
import org.apache.felix.scr.annotations.Service;

import java.util.Map;

/**
 * Created by Kusak on 7/16/2017.
 */
@Component
@Service(BeanSerializer.class)
public class BeanSerializer {
    //private ObjectMapper objectMapper;
    private Gson gson;

    @Activate
    protected void activate() {
//        SerializerFactory serializerFactory = BeanSerializerFactory
//                .instance;
//
//        objectMapper = new ObjectMapper();
//        objectMapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, true);
//        objectMapper.setSerializerFactory(serializerFactory);
        gson = new GsonBuilder()
                .excludeFieldsWithoutExposeAnnotation()
                .create();
    }

    public String convertToMap(Object bean) {
        return gson.toJson(bean);//objectMapper.convertValue(bean, Map.class);
    }
}

File: sling-react-core/src/main/kotlin/com/cathcart93/sling/core/services/BeanSerializer.kt
package com.cathcart93.sling.core.services

import com.cathcart93.sling.core.IReactController
import com.cathcart93.sling.core.ReactControllerSerializer
import com.cathcart93.sling.core.measureExecTime
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import org.apache.felix.scr.annotations.Activate
import org.apache.felix.scr.annotations.Component
import org.apache.felix.scr.annotations.Service
import org.apache.sling.api.SlingHttpServletRequest
import org.apache.sling.api.resource.Resource
import org.osgi.service.component.ComponentContext
import org.slf4j.LoggerFactory

/**
 * Created by Kusak on 7/16/2017.
 */
@Component
@Service(BeanSerializer::class)
class BeanSerializer {
    private lateinit var gson: Gson

    @Activate
    private fun init(componentContext: ComponentContext) {
        gson = GsonBuilder()
//                .setExclusionStrategies(
//                        SlingExclusionStrategy(
//                                Resource::class.java, SlingHttpServletRequest::class.java, BeanSerializer::class.java
//                        )
//                )
                .serializeNulls()
                .create()
    }

    fun convertToMap(bean: Any): String {
        val (result, time) = measureExecTime { gson.toJson(bean) }
        LOGGER.trace("Serialization time: {}", time)
        return result
    }

    companion object {
        private val LOGGER = LoggerFactory.getLogger(React::class.java)
    }
}
